### PR TITLE
Change the rules in `role.yaml` and `multiple_namespaces_role.yaml` to use the same template in `_helpers.tpl` to ensure consistency.

### PR DESCRIPTION
**Pull request by LeoLiao123**
## Why are these changes needed?

See #2233 
When the rules in role.yaml need to be modified, multiple_namespaces_role.yaml must also be updated simultaneously to continuously ensure their consistency.

## Related issue number

#2233 

## Checks

- [x] I've made sure the tests are passing.
- Testing Strategy
   - [ ] Unit tests
   - [x] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: helm-chart/kuberay-operator/templates/multiple_namespaces_role.yaml [hunk 1]
```diff
@@ -10,268 +10,6 @@ metadata:
...
-  - update
-  - watch
-- apiGroups:
-  - apiextensions.k8s.io
-  resources:
-  - customresourcedefinitions
-  verbs:
-  - get
-{{- end }}
+{{ include "role.consistencyRules" (dict "batchSchedulerEnabled" $.Values.batchScheduler.enabled) }}
```
Comment (LeoLiao123) [inline helm-chart/kuberay-operator/templates/multiple_namespaces_role.yaml:13]: The rules here and in `role.yaml` are both included from the `_helpers.tpl` template to ensure consistency.

**Comment by MortalHappiness** [comment]
LGTM

File: helm-chart/kuberay-operator/templates/_helpers.tpl [hunk 2]
```diff
@@ -54,3 +54,278 @@ Create the name of the service account to use
     {{ default "default" .Values.serviceAccount.name }}
 {{- end -}}
 {{- end -}}
+
+{{/*
+Create a template for role consistency rules.
```
Comment (kevin85421) [inline helm-chart/kuberay-operator/templates/_helpers.tpl:59]:
```suggestion
Create a template to ensure the consistency for Role and ClusterRole.
```

File: helm-chart/kuberay-operator/templates/_helpers.tpl [hunk 3]
```diff
@@ -54,3 +54,278 @@ Create the name of the service account to use
...
+*/}}
+{{- define "role.consistencyRules" -}}
```
Comment (kevin85421) [inline helm-chart/kuberay-operator/templates/_helpers.tpl:61]:
```suggestion
{{- define "role.consistentRules" -}}
```

File: helm-chart/kuberay-operator/templates/multiple_namespaces_role.yaml (hunk 1 above)
Comment (kevin85421) [inline helm-chart/kuberay-operator/templates/multiple_namespaces_role.yaml:13]:
```suggestion
{{ include "role.consistentRules" (dict "batchSchedulerEnabled" $.Values.batchScheduler.enabled) }}
```

File: helm-chart/kuberay-operator/templates/role.yaml [hunk 4]
```diff
@@ -6,279 +6,5 @@ metadata:
...
-  - update
-  - watch
-- apiGroups:
-  - apiextensions.k8s.io
-  resources:
-  - customresourcedefinitions
-  verbs:
-  - get
-{{- end }}
+{{ include "role.consistencyRules" (dict "batchSchedulerEnabled" .Values.batchScheduler.enabled) }}
```
Comment (kevin85421) [inline helm-chart/kuberay-operator/templates/role.yaml:9]:
```suggestion
{{ include "role.consistentRules" (dict "batchSchedulerEnabled" .Values.batchScheduler.enabled) }}
```

**Review by kevin85421 — COMMENTED** [review]
Can you make sure the template in `_helpers.tpl` is consistent with the old `role.yaml`? For example, we deleted `customresourcedefinitions` from role.yaml and `multiple_namespaces_role.yaml`, but I can't find it in `_helpers.tpl`.

This PR doesn't have any tests, which will make it hard for reviewers to review. It would be helpful to add details about how you manually tested it and the results.

In addition, the change is also related to batch scheduler. Can you also manual test whether https://docs.ray.io/en/latest/cluster/kubernetes/k8s-ecosystem/volcano.html works or not? Thanks!

File: helm-chart/kuberay-operator/templates/_helpers.tpl [hunk 5]
```diff
@@ -54,3 +54,284 @@ Create the name of the service account to use
...
+  - roles
+  verbs:
+  - create
+  - delete
+  - get
+  - list
+  - update
+  - watch
+- apiGroups:
+  - route.openshift.io
```
Comment (rueian) [inline helm-chart/kuberay-operator/templates/_helpers.tpl:307]: The original multiple_namespaces_role.yaml didn't contain the route.openshift.io staff. Was that a mistake and we add it in this PR to fix it?

Comment (LeoLiao123) [reply helm-chart/kuberay-operator/templates/_helpers.tpl:307]: Thank you for marking this section! I was just considering whether to directly ask a question or use a comment notation.

Comment (kevin85421) [reply helm-chart/kuberay-operator/templates/_helpers.tpl:307]: I think both `role.yaml` and `multiple_namespaces_role.yaml` should have the same permissions, so I think it is a bug fix to add `route` in `multiple_namespaces_role.yaml`.

**Comment by kevin85421** [comment]
> Can you make sure the template in `_helpers.tpl` is consistent with the old `role.yaml`? For example, we deleted `customresourcedefinitions` from role.yaml and `multiple_namespaces_role.yaml`, but I can't find it in `_helpers.tpl`.
> 
> This PR doesn't have any tests, which will make it hard for reviewers to review. It would be helpful to add details about how you manually tested it and the results.
> 
> In addition, the change is also related to batch scheduler. Can you also manual test whether https://docs.ray.io/en/latest/cluster/kubernetes/k8s-ecosystem/volcano.html works or not? Thanks!

Hi @LeoLiao123, would you mind 

(1) add details about how you manually tested it and the results to the PR description
(2) manual test whether https://docs.ray.io/en/latest/cluster/kubernetes/k8s-ecosystem/volcano.html works or not

Thanks!

**Comment by LeoLiao123** [comment]
## Make sure the template in `_helpers.tpl` is consistent with the old `role.yaml`
Hi, initially, I performed a line-by-line comparison of the `role.yaml` and `multiple_namespaces_role.yaml` files to manually test the rules. I noticed that `route.openshift.io` was missing in the original `multiple_namespaces_role.yaml`. I realized that I accidentally missed the last section when copying, but I have now added it. I also wrote a Python script to ensure that the rules in the template match those in the original `role.yaml`.

**Python Script:**
```
import re
import yaml
import pytest

# Function to clean Jinja2 template tags
def clean_jinja2_tags(yaml_content):
    cleaned_content = re.sub(r'{{-?[^{}]+-?}}', '', yaml_content)
    return cleaned_content

def extract_rules(yaml_content):
    yaml_dict = yaml.safe_load(yaml_content)
    rules = yaml_dict.get('rules', [])
    return rules

def read_file(file_path):
    with open(file_path, 'r', encoding='utf-8') as file:
        return file.read()

def test_clean_and_compare_rules():
    helpers_tpl_content = read_file('/kyberay/helm-chart/kuberay-operator/templates/_helpers.tpl')
    role_yaml_content = read_file('/kuberay/helm-chart/kuberay-operator/templates/role.yaml') # Original role.yaml 
    cleaned_role_yaml_content = clean_jinja2_tags(role_yaml_content)
    cleaned_helpers_tpl_content = clean_jinja2_tags(helpers_tpl_content)
    
    # Extract rules
    role_rules = extract_rules(cleaned_role_yaml_content)
    helpers_rules = extract_rules(cleaned_helpers_tpl_content)
    assert '{{' not in cleaned_role_yaml_content
    assert '{%' not in cleaned_role_yaml_content
    assert '{{' not in cleaned_helpers_tpl_content
    assert '{%' not in cleaned_helpers_tpl_content
    
    # Ensure extracted rules are lists and are identical
    assert isinstance(role_rules, list)
    assert isinstance(helpers_rules, list)
    # Test Consistency
    assert role_rules == helpers_rules

if __name__ == "__main__":
    pytest.main()
```
**Test Result:**
```
=========================================================================================== test session starts ===========================================================================================
platform linux -- Python 3.12.3, pytest-8.2.2, pluggy-1.5.0
rootdir: /code/kuberay
collected 1 item                                                                                                                                                                                          

helm-chart/kuberay-operator/rules_test.py .                                                                                                                                                         [100%]

============================================================================================ 1 passed in 0.05s ============================================================================================
```

## Manual test whether https://docs.ray.io/en/latest/cluster/kubernetes/k8s-ecosystem/volcano.html works or not

I have followed the steps to the final step and obtained the expected output. If you need more information to ensure there are no errors, please let me know! I will provide additional details if necessary.

**Final Output:**
```
NAME                                 READY   STATUS    RESTARTS      AGE
kuberay-operator-5dd464b777-xjwqp    1/1     Running   2 (15m ago)   16m
test-cluster-1-head-9sd97            1/1     Running   0             6m7s
test-cluster-1-worker-worker-28ccm   1/1     Running   0             6m7s
test-cluster-1-worker-worker-fhk6m   1/1     Running   0             6m7s
```

**Review by kevin85421 — APPROVED** [review]
I manually compare the rules one by one, and they are the same as the old rules.

**Pull request merged**